apps/nextjs/src/pages/admindashboard.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect } from "react";
+import { useRouter } from "next/navigation";
+import { useUser } from "@clerk/nextjs";
+
+import { api } from "~/utils/api";
+
+export default function AdminDashboard() {
+  const { user, isLoaded } = useUser();
+  const router = useRouter();
+  const allUsersQuery = api.user.getAllUsers.useQuery();
+  const allClerkUsersQuery = api.user.getAllClerkUsers.useQuery();
+
+  const changeApprovalStatusMutation =
+    api.user.changeApprovalStatus.useMutation();
+
+  useEffect(() => {
+    // user is not loaded, we're not sure if they're an admin yet
+    if (!isLoaded || !user) {
+      return;
+    }
+
+    if (user.publicMetadata.isAdmin !== true) {
+      router.push("/");
+    }
```

Comment:
can you add this check to the middleware instead?

Reply:
moved